src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -240,18 +240,16 @@ private HeadersPayload ConstructHeadersPayload(BlockLocator locator, uint256 has
         /// the tip of the best chain we think the peer has.
         /// </para>
         /// </remarks>
-        private async Task ProcessHeadersAsync(INetworkPeer peer, HeadersPayload headersPayload)
+        protected async Task ProcessHeadersAsync(INetworkPeer peer, List<BlockHeader> headers)
```

Comment:
You are going to have conflicts with @kogot add an override here.

Reply:
Yeah. I think @kogot will be able to reuse changes I did to CMB since a lot of his things need to override same logic. So it will actually help him

Reply:
@kogot check out this class how it uses inheritance as much as possible (to do the same for PH).